python/lsst/daf/butler/registry/datasets/byDimensions/_manager.py

```diff
@@ -115,6 +115,22 @@ def refresh(self, *, universe: DimensionUniverse) -> None:
         self._byName = byName
         self._byId = byId
 
+    def deleteDatasetType(self, name: str) -> None:
+        # Docstring inherited from DatasetRecordStorageManager.
+        compositeName, componentName = DatasetType.splitDatasetTypeName(name)
+        if componentName is not None:
+            raise ValueError(f"Cannot delete a dataset type of a component of a composite (given {name})")
+        storage = self._byName.get(compositeName)
+        if storage is None:
```

Comment:
Should this return silently?

Reply:
Given that registration is a no-op if the dataset already exists, I suspect we will want deletion to be a no-op if it's already deleted.